### PR TITLE
adding IFFE docs

### PR DESCRIPTION
**Pull request by ritter-eb**
adding docs for IFFE anti-pattern (at least in ES5 and above) as per @gagoar request

### CODE REVIEW

File: general/README.md [hunk 1]
```diff
@@ -249,6 +250,42 @@ if (isAllowed) {
...
 **[⬆ back to top](#table-of-contents)**
 
+## Functions
+
+### Immediately-invoked function expressions
+
+Avoid using IFFEs (immediately-invoked function expressions) if possible. The `bind` function is preferred when dealing with variable scope in closures (note that `bind` is only available in ES5+).
+
+```js
+// good (using .bind)
```
Comment (gagoar) [inline general/README.md]: we should put something more generic in the function. maybe doing something more trivial ?

File: general/README.md [hunk 2]
```diff
@@ -249,6 +250,42 @@ if (isAllowed) {
...
+logCheckpoint = function(appOptions) {
+    janus.logCheckpointViaAjax('google_translate_dropdown_changed', {
+        'event_id': appOptions.eventId
+    });
+};
+
+$('.goog-te-combo').change(logCheckpoint.bind(null, appOptions));
```
Comment (gagoar) [inline general/README.md]: the selector should have a variable holding the string.

**Comment by gagoar** [comment]
this looks great. some comments tho.

**Comment by ritter-eb** [comment]
@gagoar comment changes made!

**Comment by gagoar** [comment]
I will squash the commits, and add that we currently use `es5-shim` so bind is there by default.

File: general/README.md [hunk 3]
```diff
@@ -249,6 +250,46 @@ if (isAllowed) {
...
 **[⬆ back to top](#table-of-contents)**
 
+## Functions
+
+### Immediately-invoked function expressions
+
+Avoid using IFFEs (immediately-invoked function expressions) if possible. The `bind` function is preferred when dealing with variable scope in closures (note that `bind` is only available in ES5+).
+
+```js
+var purchaseTicketsButton = '.goog-te-combo';
```
Comment (gagoar) [inline general/README.md]: `purchaseTicketButtonSelector`

File: general/README.md [hunk 4]
```diff
@@ -249,6 +250,46 @@ if (isAllowed) {
...
+
+// good (using .bind)
+buyTicketFunction = function(options) {
+    buyTickets({
```
Comment (gagoar) [inline general/README.md]: you could do `buyTickets(_.pick(options, 'eventId', 'ticketId'));`

File: general/README.md [hunk 5]
```diff
@@ -249,6 +250,46 @@ if (isAllowed) {
...
+        'ticket_id': options.ticketId
+    });
+};
+
+$(purchaseTicketsButton).click(buyTicketFunction.bind(null, options));
+
+// bad (using IFFEs)
+buyTicketFunction = (function(options) {
+    return function() {
+        buyTickets({
```
Comment (gagoar) [inline general/README.md]: same here with `_.pick`

**Pull request merged**